File: IT/1/Variabili, operatori, strutture di controllo/Java/SwitchTest.java
/* file SwitchTest.java */

public class SwitchTest {
	public static void main(String[] args) {
		String s = System.console().readLine("Inserisci un intero compreso tra 1 e 3: ");

		if (!s.matches("[\\d]+")) {
			System.out.println("Numero non valido!");
			return;
		}

		int n = Integer.parseInt(s);

		switch (n) {
			case 1: 
				System.out.println("Hai digitato il numero 1.");
				break;

			case 2: 
				System.out.println("Hai digitato il numero 2.");
				break;

			case 3: 
				System.out.println("Hai digitato il numero 3.");
				break;

			default: 
				System.out.println("Numero fuori dell'intervallo!");
				break;
		}
	}
}

File: IT/1/Variabili, operatori, strutture di controllo/C++/switchTest.cpp
/*
file switchTest.cpp 
compilare con: g++ -std=c++11 switchTest.cpp
*/

#include <iostream>

using std::cin;
using std::cout;
using std::endl;
using std::string;

int main()
{
	cout << "Inserisci un intero positivo compreso tra 1 e 3: ";
	string s;
	getline(cin, s);

	if (s.find_first_not_of("0123456789") != string::npos) {
		cout << "Numero non valido!" << endl;
		return 0;
	}

	int n = stoi(s);

	switch (n) {
		case 1:
			cout << "Hai digitato il numero 1." << endl;
			break;

		case 2: 
			cout << "Hai digitato il numero 2." << endl;
			break;

		case 3:
			cout << "Hai digitato il numero 3." << endl;
			break;
			
		default:
			cout << "Numero fuori dell'intervallo!" << endl;
			break;
	}
}

